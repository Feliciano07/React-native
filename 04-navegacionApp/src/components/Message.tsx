import React from 'react'
import { Text, View } from 'react-native'

type ComponentsProps = {
    mensaje?: string
}

const Message = (props:ComponentsProps ) => {

    return(
        <View>
            <Text>
                {props.mensaje}
            </Text>
        </View>
    )
}

export default Message