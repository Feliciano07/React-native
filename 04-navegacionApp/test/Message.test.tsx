import React from 'react';
import { render } from '@testing-library/react-native';
import Message from '../src/components/Message';




describe('Testing of Message', () => {
	it('renders correctly', () => {
		const { getByText } = render(<Message />);
		const helloText = getByText('');
		expect(helloText).toBeTruthy();
	});

	it('render correctly with props mensaje ', () => {
		const mensaje = 'Mi prueba'
		const { getByText } = render(<Message mensaje={mensaje} />);
		const helloText = getByText(mensaje);
		expect(helloText).toBeTruthy();
	});

	it('render correctly with props mensaje change ', () => {
		const mensaje = 'Mi prueba'
		const { getByText, rerender } = render(<Message mensaje={mensaje} />);
		const helloText = getByText(mensaje);
		expect(helloText).toBeTruthy();

		// Rerender the component with a different message
		const nuevoMensaje = 'Â¡Hola a todos!';
		rerender(<Message mensaje={nuevoMensaje} />);
		const nuevoMensajeText = getByText(nuevoMensaje);
		expect(nuevoMensajeText).toBeTruthy();
	})

	it('matches snapshot', () => {
		const { toJSON } = render(<Message />);
		expect(toJSON()).toMatchSnapshot();
	});
});