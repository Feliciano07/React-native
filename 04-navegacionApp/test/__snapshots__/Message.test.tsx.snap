// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing of Message matches snapshot 1`] = `
<View>
  <Text />
</View>
`;
